examples/Cpp/PrimitiveRestart/Example.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ * Example.cpp (Example_PrimitiveRestart)
+ *
+ * This file is part of the "LLGL" project (Copyright (c) 2015-2019 by Lukas Hermanns)
+ * See "LICENSE.txt" for license information.
+ */
+
+#include <ExampleBase.h>
+#include <vector>
+
+class Example_PrimitiveRestart : public ExampleBase
+{
+    LLGL::ShaderProgram*    shaderProgram       = nullptr;
+    LLGL::PipelineState*    pipeline            = nullptr;
+    LLGL::Buffer*           vertexBuffer        = nullptr;
+    LLGL::Buffer*           indexBuffer16       = nullptr;
+    LLGL::Buffer*           indexBuffer32       = nullptr;
+
+    size_t                  indexCount16        = 0;
+    size_t                  indexCount32        = 0;
+
+public:
+
+    Example_PrimitiveRestart() :
+    ExampleBase { L"LLGL Example: PrimitiveRestart" }
+    {
+        auto vertexFormats = CreateBuffers();
+        shaderProgram = LoadStandardShaderProgram(vertexFormats);
+        CreatePipeline();
+    }
+
+private:
+
+    // Vertex data structure
+    struct Vertex
+    {
+        Gs::Vector2f        position;
+        LLGL::ColorRGBAub   color;
+    };
+
+    template <typename T>
+    static void addSquare(float centerX, float centerY, float size, std::vector<Vertex>& vertices, std::vector<T>& indices)
+    {
+        float left = centerX - size / 2.0;
+        float right = centerX + size / 2.0;
+        float top = centerY + size / 2.0;
+        float bottom = centerY - size / 2.0;
+
+        size_t startIndex = vertices.size();
+        vertices.push_back({{ right, top    }, { 255,   0,   0, 255 } });
+        vertices.push_back({{ right, bottom }, {   0, 255,   0, 255 } });
+        vertices.push_back({{ left,  top    }, {   0,   0, 255, 255 } });
+        vertices.push_back({{ left,  bottom }, { 255, 255, 255, 255 } });
+
+        if (!indices.empty())
+            indices.push_back((T) 0xFFFFFFFF);
```

Comment:
Please use the constructor expression here, i.e. `T(0xFFFFFFFF)`. I'm not a fan of C-style casts as they are error prone and nasty to search for. I don't mind if they are in any of the third party dependencies though. In this case it doesn't really matter, but I'd like to avoid them completely.